tests/reset/reset.cpp

```diff
@@ -276,65 +204,176 @@ TEST(Reset, printResetWithWhen)
+    const std::string e2 =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\">\n"
+        "  <component name=\"valid_name\">\n"
+        "    <variable name=\"variable1\"/>\n"
+        "    <variable name=\"variable2\"/>\n"
+        "    <reset variable=\"variable1\" test_variable=\"variable2\" order=\"2\">\n" // only difference is order of reset
+        "      <test_value>\n"
+        "        <math xmlns=\"http://www.w3.org/1998/Math/MathML\"/>\n"
+        "      </test_value>\n"
+        "      <reset_value>\n"
+        "        <math xmlns=\"http://www.w3.org/1998/Math/MathML\"/>\n"
+        "      </reset_value>\n"
+        "    </reset>\n"
+        "  </component>\n"
+        "</model>\n";
+    const std::string e3 =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\">\n"
+        "  <component name=\"valid_name\">\n"
+        "    <variable name=\"variable1\"/>\n"
+        "    <variable name=\"variable2\"/>\n"
+        "  </component>\n"
+        "</model>\n";
+
     libcellml::ModelPtr m = createModelWithComponent();
     libcellml::ComponentPtr c = m->component(0);
-    libcellml::ResetPtr r = std::make_shared<libcellml::Reset>();
+    libcellml::VariablePtr v1 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v2 = std::make_shared<libcellml::Variable>();
+    libcellml::ResetPtr r1 = std::make_shared<libcellml::Reset>();
+    libcellml::ResetPtr r2 = std::make_shared<libcellml::Reset>();
+    libcellml::ResetPtr r3 = std::make_shared<libcellml::Reset>();
+    libcellml::Printer printer;
 
-    libcellml::WhenPtr w1 = std::make_shared<libcellml::When>();
-    libcellml::WhenPtr w2 = std::make_shared<libcellml::When>();
-    libcellml::WhenPtr w3 = std::make_shared<libcellml::When>();
+    c->setName(in);
+    v1->setName("variable1");
+    v2->setName("variable2");
 
-    r->addWhen(w1);
-    r->addWhen(w2);
-    r->addWhen(w3);
-    c->addReset(r);
+    r1->setVariable(v1);
+    r1->setTestVariable(v2);
+    r1->setOrder(1);
+    r1->setResetValue(EMPTY_MATH);
+    r1->setTestValue(EMPTY_MATH);
 
-    libcellml::Printer p;
+    c->addReset(r1);
+    c->addVariable(v1);
+    c->addVariable(v2);
 
-    const std::string a = p.printModel(m);
-    EXPECT_EQ(e, a);
+    a = printer.printModel(m);
+    EXPECT_EQ(e1, a);
+
+    // Add another reset
+    r2->setVariable(v1);
+    r2->setTestVariable(v2);
+    r2->setOrder(2);
+    r2->setResetValue(EMPTY_MATH);
+    r2->setTestValue(EMPTY_MATH);
+    c->addReset(r2);
+
+    // Remove the first one and print the model
+    c->removeReset(r1);
+    a = printer.printModel(m);
+    EXPECT_EQ(e2, a);
+
+    // Remove the second one and print the model
+    c->removeReset(0);
+    a = printer.printModel(m);
+    EXPECT_EQ(e3, a);
+
+    // Add them both back in and use removeAllResets to remove them
+    c->addReset(r1);
+    c->addReset(r2);
```

Comment:
Probably worth checking that they have indeed been added back...? I mean, should `addReset()` not working, `removeAllResets()` would clearly have nothing to do and `EXPECT_EQ(e3, a)` would necessarily be true.